aws_requests_auth/tests/test_boto_utils.py

```diff
@@ -0,0 +1,45 @@
+import os
+import unittest
+
+import mock
+
+from aws_requests_auth.boto_utils import BotoAWSRequestsAuth, get_credentials
+
+
+class TestBotoUtils(unittest.TestCase):
+    """
+    Tests for boto_utils module.
+    """
+
+    def setUp(self):
+        self.saved_env = {}
+        for var in ['AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'AWS_SESSION_TOKEN']:
+            self.saved_env[var] = os.environ.get(var)
+            os.environ[var] = 'test-%s' % var
```

Comment:
neat -- I don't have much experience mutating environment variables during the course of a test.

is the following understanding correct?

- the mutations we make here should only affect the env variables for the python process running our tests
- we are being a good citizen in undoing our changes to the env in `tearDown()` 
     - i.e. we could conceivably get away without undoing the env variable changes; we'd just assume that the python process running our tests will get thrown out and nobody will care that we screwed with the environment variables?

Reply:
Yes - correct (AFAIK, at least) on both counts! Good questions.

I of course use `aws-requests-auth` to retrieve temporary credentials from EC2 instance meta data, not just from environment variables which would be easy enough to pass in manually, but `botocore` conveniently checks environment variables too, so those make a worthy substitute for testing purposes.